src/library/scala/collection/mutable/HashMap.scala

```diff
@@ -525,6 +526,22 @@ class HashMap[K, V](initialCapacity: Int, loadFactor: Double)
   override def mapFactory: MapFactory[HashMap] = HashMap
 
   override protected[this] def stringPrefix = "HashMap"
+
+  override def hashCode: Int = {
+    val mapIterator = this.iterator
+    if (mapIterator.isEmpty) MurmurHash3.emptyMapHash
```

Comment:
Could we use the size instead of creating an iterator?
```scala
if (size == 0) MurmurHash3.emptyMapHash
else ...
```

Reply:
Calling `iterator` on a empty collection are pretty much allocation free throughout 2.13.x collections so there a performance difference.

Reply:
but you're not using it when it's not empty

Reply:
You're right, just calling `this.isEmpty` would be simpler (if not appreciably faster.) I'll update.